demo/java/src/ai/picovoice/picovoicedemo/FileDemo.java

```diff
@@ -0,0 +1,270 @@
+            if (inference.getIsUnderstood()) {
+
+                System.out.println("{");
+                System.out.println(String.format("  intent : '%s'", inference.getIntent()));
+                System.out.println("  slots : {");
+                for (Map.Entry<String, String> slot : inference.getSlots().entrySet()) {
+                    System.out.println(String.format("    %s : '%s'", slot.getKey(), slot.getValue()));
+                }
+                System.out.println("  }");
+                System.out.println("}");
+            } else {
+                System.out.println("Didn't understand the command.");
+            }
+        };
+
+        Picovoice picovoice = null;
+        try {
+            picovoice = new Picovoice.Builder()
+                    .setKeywordPath(keywordPath)
+                    .setWakeWordCallback(wakeWordCallback)
+                    .setContextPath(contextPath)
+                    .setInferenceCallback(inferenceCallback)
+                    .setPorcupineLibraryPath(porcupineLibraryPath)
+                    .setPorcupineModelPath(porcupineModelPath)
+                    .setPorcupineSensitivity(porcupineSensitivity)
+                    .setRhinoLibraryPath(rhinoLibraryPath)
+                    .setRhinoModelPath(rhinoModelPath)
+                    .setRhinoSensitivity(rhinoSensitivity)
+                    .build();
+
+            AudioFormat audioFormat = audioInputStream.getFormat();
+
+            if (audioFormat.getSampleRate() != (float)picovoice.getSampleRate() || audioFormat.getSampleSizeInBits() != 16) {
+                throw new IllegalArgumentException(String.format("Invalid input audio file format. " +
+                        "Input file must be a %dkHz, 16-bit audio file.", picovoice.getSampleRate()));
+            }
+
+            if (audioFormat.getChannels() > 1) {
+                System.out.println("Picovoice processes single-channel audio, but a multi-channel file was provided. " +
+                        "Processing leftmost channel only.");
+            }
+
+            int frameIndex = 0;
+            short[] picovoiceFrame = new short[picovoice.getFrameLength()];
+
+            ByteBuffer sampleBuffer = ByteBuffer.allocate(audioFormat.getFrameSize());
+            sampleBuffer.order(ByteOrder.LITTLE_ENDIAN);
+            while (audioInputStream.available() != 0) {
+
+                int numBytesRead = audioInputStream.read(sampleBuffer.array());
+                if (numBytesRead < 2) {
+                    break;
+                }
+
+                picovoiceFrame[frameIndex++] = sampleBuffer.getShort(0);
+
+                if (frameIndex == picovoiceFrame.length) {
+
+                    picovoice.process(picovoiceFrame);
+                    frameIndex = 0;
+                }
+            }
+        } catch (Exception e) {
+            System.out.println(e.toString());
+        } finally {
+            if (picovoice != null) {
+                picovoice.delete();
```

Comment:
add "picovoice = null" after. It's redundant but it communicates what's happening